pkg/gpu/stream.go

```diff
@@ -67,17 +80,68 @@ type kernelSpan struct {
 
 	// numKernels is the number of kernels that were launched during the span
 	numKernels uint64
+
+	// avgMemoryUsage is the average memory usage during the span, per allocation type
+	avgMemoryUsage map[model.MemAllocType]uint64
 }
 
-func newStreamHandler() *StreamHandler {
+func newStreamHandler(key streamKey, sysCtx *systemContext) *StreamHandler {
 	return &StreamHandler{
 		memAllocEvents: make(map[uint64]gpuebpf.CudaMemEvent),
+		key:            key,
+		sysCtx:         sysCtx,
 	}
 }
 
 func (sh *StreamHandler) handleKernelLaunch(event *gpuebpf.CudaKernelLaunch) {
-	// Copy events, as the memory can be overwritten in the ring buffer after the function returns
-	sh.kernelLaunches = append(sh.kernelLaunches, *event)
+	enrichedLaunch := &enrichedKernelLaunch{
+		CudaKernelLaunch: *event, // Copy events, as the memory can be overwritten in the ring buffer after the function returns
+	}
+
+	err := sh.tryAttachKernelData(enrichedLaunch)
+	if err != nil {
+		log.Warnf("Error attaching kernel data: %v", err)
```

Comment:
```suggestion
		log.Warnf("error attaching kernel data: %w", err)
```

Reply:
Is that needed here? I thought in log messages it's better to use `%v`